design/src/docs/anatomy.md

```diff
@@ -0,0 +1,601 @@
+# The Anatomy of a Service
+
+What is [Smithy](https://awslabs.github.io/smithy/2.0/index.html)? At a high-level, it's a grammar for specifying services modulo business logic. A [Smithy Service](https://awslabs.github.io/smithy/2.0/spec/service-types.html#service) specifies a set of function signatures in the form of [Operations](https://awslabs.github.io/smithy/2.0/spec/service-types.html#operation) to which encapsulate business logic. A Smithy implementation should, for each Smithy Service, provide a builder, which accepts functions conforming to said signatures, and returns a service subject to the semantics specified by the model.
+
+This is an dissection of the various mechanisms at work inside Smithy Rust in order to provide such a builder. This is intended for new contributors or users who are interested in the internal details.
```

Comment:
This should be changed to emphasize that this is not a description of the logic/implementation details of the Kotlin code but rather a description of how the Rust composes together and it's relation to the Smithy model.